pkg/config/config.go

```diff
@@ -307,34 +314,47 @@ func getenv(name string) string {
 // to the flagset before parsing them with the values specified by
 // args.
 func Load(fs *flag.FlagSet, args []string) (*Config, error) {
-	return load(fs, args, func(url string, expand bool, c *Config) error {
-		if features.Enabled(fs, featRemoteConfigs) {
-			return LoadRemote(url, expand, c)
-		}
-		if features.Enabled(fs, featDynamicConfig) && !features.Enabled(fs, featIntegrationsNext) {
-			return fmt.Errorf("integrations-next must be enabled for dynamic configuration to work")
-		} else if features.Enabled(fs, featDynamicConfig) {
-			return LoadDynamicConfiguration(url, expand, c)
+	return load(fs, args, func(path, fileType string, expandArgs bool, c *Config) error {
+		switch fileType {
+		case fileTypeYAML:
+			if features.Enabled(fs, featRemoteConfigs) {
+				return LoadRemote(path, expandArgs, c)
+			}
+			return LoadFile(path, expandArgs, c)
+		case fileTypeDynamic:
+			if !features.Enabled(fs, featDynamicConfig) {
+				return fmt.Errorf("feature %q must be enabled to use file type %s", featDynamicConfig, fileTypeDynamic)
+			} else if !features.Enabled(fs, featIntegrationsNext) {
+				return fmt.Errorf("feature %q must be enabled to use file type %s", featIntegrationsNext, fileTypeDynamic)
+			} else if features.Enabled(fs, featRemoteConfigs) {
+				return fmt.Errorf("feature %q can not be enabled with file type %s", featRemoteConfigs, fileTypeDynamic)
+			} else if expandArgs {
+				return fmt.Errorf("-config.expand-env can not be used with file type %s", fileTypeDynamic)
+			}
```

Comment:
Note that this introduces some extra validations that didn't exist before, where unexpected combinations of flags would be silently ignored (cc @mattdurham)